crates/red_knot/src/semantic/types.rs

```diff
@@ -764,16 +764,49 @@ pub(crate) struct UnionType {
 
 impl UnionType {
     fn display(&self, f: &mut std::fmt::Formatter<'_>, store: &TypeStore) -> std::fmt::Result {
-        f.write_str("(")?;
+        let (int_literals, other_types): (Vec<Type>, Vec<Type>) = self
+            .elements
+            .iter()
+            .copied()
+            .partition(|ty| matches!(ty, Type::IntLiteral(_)));
+        let num_types = usize::from(!int_literals.is_empty()) + other_types.len();
+        if num_types > 1 {
+            f.write_str("(")?;
```

Comment:
Why are we parenthesizing any of these displays, OOI? `Literal[1] | None` feels just as clear to me as `(Literal[1] | None)`

Reply:
Yeah, I was doing this for clarity in case the union was embedded in other unions/intersections, but I think a) we should always normalize to disjunctive normal form (union of intersections), and b) even there, we should only add parens contextually as needed for clarity, not always. So I'll remove this.